xbmc/addons/interfaces/Filesystem.h

```diff
@@ -42,12 +54,12 @@ struct Interface_Filesystem
   static bool get_directory(void* kodiBase,
                             const char* path,
                             const char* mask,
-                            VFSDirEntry** items,
+                            struct VFSDirEntry** items,
                             unsigned int* num_items);
-  static void free_directory(void* kodiBase, VFSDirEntry* items, unsigned int num_items);
+  static void free_directory(void* kodiBase, struct VFSDirEntry* items, unsigned int num_items);
 
   static bool file_exists(void* kodiBase, const char* filename, bool useCache);
-  static int stat_file(void* kodiBase, const char* filename, struct __stat64* buffer);
+  static bool stat_file(void* kodiBase, const char* filename, struct STAT_STRUCTURE* buffer);
```

Comment:
Is there another opportunity to change a stat's `int` return type to `bool`? I saw a comparison of a `Stat()` function against `0` above. Just wondering.

Reply:
About them I change it complete, as `void*` is bad for other languages :smile:.
Make it now with them:
```cpp
    bool (*io_control_seek_possible)(void* kodiBase, void* file);
    bool (*io_control_get_cache_status)(void* kodiBase, void* file, struct VFS_IOCTRL_CACHE_STATUS_DATA* status);
    bool (*io_control_set_cache_rate)(void* kodiBase, void* file, unsigned int rate);
    bool (*io_control_set_retry)(void* kodiBase, void* file, bool retry);
```